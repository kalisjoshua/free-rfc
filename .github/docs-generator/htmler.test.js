const {htmler} = require("./htmler")

describe("htmler", () => {
  // {html, id, topics, type}
  it("should return html content with \"unknown\" filed in for all template variables", () => {
    expect(htmler()).toMatchSnapshot()
  })

  it("should return html content with \"unknown\" filed in for all template variables", () => {
    const data = {
      allRFCs: [
        {
          html: "Mock RFC article.",
          id: "mock-1",
          topics: ["mock", "test"],
          type: "mock",
        }
      ],
      pageTitle: "Mock Title",
      SHA: "12345",
      version: "m.o.c-k",
    }

    expect(htmler(data)).toMatchSnapshot()
  })
})
