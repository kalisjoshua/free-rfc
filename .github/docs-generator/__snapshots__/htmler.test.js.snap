// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmler should return html content with "unknown" filed in for all template variables 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"/><meta charset=\\"utf-8\\"/><meta name=\\"viewport\\"content=\\"width=device-width,initial-scale=1\\"/><meta name=\\"description\\"content=\\"parent git commit: undefined\\"/><meta name=\\"description\\"content=\\"Lean RFC version: undefined\\"/><link href=\\"./chimera.ico\\"rel=\\"icon\\"/><link href=\\"./chimera.png\\"rel=\\"icon\\"type=\\"image/png\\"/><link href=\\"./chimera.svg\\"rel=\\"icon\\"type=\\"image/svg+xml\\"/><title>undefined</title><link href=\\"./rfc-emcee.css\\"rel=\\"stylesheet\\"/><script defer src=\\"./rfc-emcee.mjs\\"type=\\"module\\"></script><body id=\\"top\\"><a class=\\"skip-to-main\\"href=\\"#main\\">Skip to main content</a><div class=\\"layout\\"><header role=\\"banner\\"><h1>undefined</h1></header><div data-app></div></div>"`;

exports[`htmler should return html content with "unknown" filed in for all template variables 2`] = `"<!DOCTYPE html><html lang=\\"en-US\\"/><meta charset=\\"utf-8\\"/><meta name=\\"viewport\\"content=\\"width=device-width,initial-scale=1\\"/><meta name=\\"description\\"content=\\"parent git commit: 12345\\"/><meta name=\\"description\\"content=\\"Lean RFC version: m.o.c-k\\"/><link href=\\"./chimera.ico\\"rel=\\"icon\\"/><link href=\\"./chimera.png\\"rel=\\"icon\\"type=\\"image/png\\"/><link href=\\"./chimera.svg\\"rel=\\"icon\\"type=\\"image/svg+xml\\"/><title>Mock Title</title><link href=\\"./rfc-emcee.css\\"rel=\\"stylesheet\\"/><script defer src=\\"./rfc-emcee.mjs\\"type=\\"module\\"></script><body id=\\"top\\"><a class=\\"skip-to-main\\"href=\\"#main\\">Skip to main content</a><div class=\\"layout\\"><header role=\\"banner\\"><h1>Mock Title</h1></header><div data-app></div><article data-topics=\\"mock,test\\"data-type=\\"mock\\"id=\\"mock-1\\">Mock RFC article.</article></div>"`;
